src/postgres-data/buildUtilities.ts

```diff
@@ -11,127 +11,99 @@
-    const data = getData()
-    if (data.projectID && data.manuscriptID) {
-      api.saveProject(data.projectID, onlyProjectModels)
+    if (!model._id) {
+      throw new Error('Model ID required')
     }
-  }
-
-  const bulkPersistentManuscriptSave = (models: ManuscriptModel[]) => {
-    const clearedModels = models.filter((model) => {
-      return model.objectType !== ObjectTypes.Project
-    })
 
-    const data = getData()
-
-    if (data.projectID && data.manuscriptID) {
-      return api
-        .saveProject(data.projectID, clearedModels)
-        .then(() => {
-          return true // not sure what will be returned at this point
-        })
-        .catch((e) => {
-          return false
-        })
-    } else {
-      return Promise.reject(false)
+    if (manuscriptIDTypes.has(model.objectType)) {
+      containerIDs.manuscriptID = manuscriptID
     }
-  }
 
-  const saveModel = async <T extends Model>(
-    model: T | Build<T> | Partial<T>
-  ) => {
-    if (!model._id) {
-      throw new Error('Model ID required')
+    return {
+      ...model,
+      ...containerIDs,
     }
+  }
 
-    const data = getData()
-    if (!data.modelMap || !data.manuscriptID || !data.projectID) {
-      throw new Error(
-        'State misses important element. Unable to savel a model.'
+  const saveProject = async (models: Model[], projectID: string) => {
+    try {
+      const filtered = models.filter(
+        (m) => m.objectType !== ObjectTypes.Project
       )
+      await api.saveProject(projectID, filtered)
+      return true
+    } catch (e) {
+      return false
     }
+  }
 
-    const containedModel = model as T & ContainedProps
-
-    // NOTE: this is needed because the local state is updated before saving
-    const containerIDs: ContainerIDs = {
-      containerID: data.projectID,
+  const getModel = <T extends Model>(id: string) => {
+    const state = getState()
+    if (!state.modelMap) {
+      return
     }
+    return state.modelMap.get(id) as T | undefined
+  }
 
-    if (isManuscriptModel(containedModel)) {
-      containerIDs.manuscriptID = data.manuscriptID
-    }
+  const saveModels = async (
+    models: Model[] | Build<Model>[] | Partial<Model>[],
+    excludeIDs?: Set<string>
+  ) => {
+    const state = getState()
+    const projectID = state.projectID
+    const manuscriptID = state.manuscriptID
 
-    const newModel = {
-      ...containedModel,
-      ...containerIDs,
+    if (!state.modelMap || !manuscriptID || !projectID) {
+      throw new Error('Unable to save due to incomplete data')
     }
 
-    const modelMap = new Map(data.modelMap)
-    modelMap.set(containedModel._id, newModel)
+    const modelMap = new Map<string, Model>()
 
-    // data.modelMap.set(containedModel._id, newModel)
+    for (const [id, model] of state.modelMap) {
+      const type = model.objectType as ObjectTypes
+      if (!excludeTypes.has(type) && !excludeIDs?.has(id)) {
```

Comment:
@mnatsheh with this check we are not going to save manuscript model! and if you recall `/project/../manuscript/..` after save will not get `MPManuscript`. so should we use a different endpoint to get manuscript?

Reply:
@asouqi , we should update this to save the manuscript. Feel free to open a PR or file a ticket and I'll take a look.